dali/base/dafdesc.cpp

```diff
@@ -3145,6 +3145,166 @@ static void setupContainerizedStorageLocations()
     }
 }
 
+struct GroupInformation : public CInterface
+{
+public:
+    GroupInformation(const char * _name) : name(_name) {}
+
+    unsigned ordinality() const { return hosts.ordinality(); }
+
+    bool checkIsSubset(const GroupInformation & other); // save information if it is a subset of other
+    void createStoragePlane(IPropertyTree * storage, unsigned copy) const;
+
+public:
+    StringBuffer name;
+    StringBuffer dir;
+    StringArray hosts;
+    const GroupInformation * container = nullptr;
+    unsigned containerOffset = 0;
+    GroupType groupType = grp_unknown;
+    bool dropZone = false;
+};
+
+using GroupInfoArray = CIArrayOf<GroupInformation>;
+
+bool GroupInformation::checkIsSubset(const GroupInformation & other)
+{
+    //Find the first ip that matches, and then check the next ips within the other list also match
+    unsigned thisSize = hosts.ordinality();
+    unsigned otherSize = other.hosts.ordinality();
+    for (unsigned i=0; i <= otherSize-thisSize; i++)
```

Comment:
think you need to check that otherSize is not < thisSize before here and reject.

Reply:
That can never happen but I'll add an assert to check